src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerBase.cs

```diff
@@ -267,6 +289,82 @@ public override Task RemoveFromGroupAsync(string connectionId, string groupName,
             return WriteAckableMessageAsync(message, cancellationToken);
         }
 
+#if NET7_0_OR_GREATER
+        public override async Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object[] args, CancellationToken cancellationToken = default)
+        {
+            if (IsInvalidArgument(connectionId))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(connectionId));
+            }
+
+            if (IsInvalidArgument(methodName))
+            {
+                throw new ArgumentException(NullOrEmptyStringErrorMessage, nameof(methodName));
+            }
+            // globally distinct invocationId
+            // $"{connectionId}{_callerId}{_clientResults.GetNewInvocation()}";
+            var invocationId = _clientInvocationManager.Caller.GetNewInvocationId(connectionId, _serverGUID);
+            var task = _clientInvocationManager.Caller.AddInvocation<T>(connectionId, invocationId, cancellationToken);
```

Comment:
how T is set for this task? Isn't it always wait for completionMessage to set the result?